### DISCUSSION TITLE
this.third.physics.add   properties for glb objects to make them float in collision with object that has bigger mass

### DISCUSSION THREAD

**Discussion by lv88h** [enable3d/enable3d#300] [Help / Questions]
I have two objects - one is floating red box that floats around when collided. Other is .GLB object "chair". .GLB object for some reason when collided only works as a "wall" , so it is not floating but static, and can't be rolled around like a football. 

this.third.physics.add.box({ y: 10, x: 754.2014 }, { lambert: { color: 'red' } })

          
             this.third.load.gltf('/assets/glb/chair.glb').then(object => {
            const scene2 = object.scenes[0]

            const chair = new ExtendedObject3D()
            console.log("chair", chair);
            chair.name = 'scene'
            chair.add(scene2)			
            chair.position.set(634.2014, 1, -250.63472)
            this.third.add.existing(chair)
			this.third.physics.add.existing(chair, {
                    shape: 'concave',
                    mass: 1,
                    collisionFlags: 1,
                    autoCenter: false
            })
                  this.third.physics.add(chair);
                  
                  
            //chair.body.setBounciness(0.5);
            //// add animations
            //// sadly only the flags animations works
            //object.animations.forEach((anim2, i2) => {
              //chair.mixer = this.third.animationMixers.create(chair)
              //// overwrite the action to be an array of actions
              //chair.action = []
              //chair.action[i2] = book.mixer.clipAction(anim2)
              //chair.action[i2].play()
            //})

            chair.traverse(child2 => {
              if (child2.isMesh) {
                child2.castShadow = child2.receiveShadow = false
                child2.material.metalness = 0
                child2.material.roughness = 1

                if (/mesh/i.test(child2.name)) {
                  this.third.physics.add.existing(child2, {
                    shape: 'concave',
                    mass: 1,
                    collisionFlags: 1,
                    autoCenter: false
                  })
                  child2.body.setAngularFactor(0, 0, 0)
                  child2.body.setLinearFactor(0, 0, 0)
                }
              }
            })
            
            
            
          })

**Comment by yandeu** [comment]
You have set `.setAngularFactor(0, 0, 0)` and `.setLinearFactor(0, 0, 0)`.
Means your body can't move nor rotate.

**Reply by lv88h** [reply]
I commented those lines, and it still can't move and rotate, should they be changed differently?

**Reply by yandeu** [reply]
Try `convex` body instead of `concave`.

Have a look at this example:
https://github.com/enable3d/enable3d-website/blob/21b1bbc7af9a24513670d2fc827c741b577cdb46/src/examples/gltf-loader-with-physics.html#L50

**Reply by lv88h** [reply]
Can I guess where is documentation for this? It did not work, but this code should be somehow written...

*oh never mind You did some link...

**Reply by lv88h** [reply]
well I tried to use the whole code example template downloaded too, and seems like suzanne is not appearing on the box....
I had totally rewrite libs etc too just to get the base of the field...

**Reply by lv88h** [reply]
https://pastebin.pl/view/55a73099

**Reply by lv88h** [reply]
https://ibb.co/26WBrDt

**Reply by yandeu** [reply]
So it works? If not please, make your repository public so I can help you with the code.

**Reply by lv88h** [reply]
it does not, as You can see, no "suzzane" has been spawned on plane.... well I give You the game code I've written so far, just so You can get what I did originally (in link, no repo) - https://pastebin.pl/view/d0a6cce6

**Reply by yandeu** [reply]
I'm sorry. Your code is a mess. I don't even understand what you are trying to achieve.

Please have a closer look at the enable3d and three.js examples.

**Comment by lv88h** [comment]
This does not work. Make sure it does.

Yannick ***@***.***> (šajā datumā: Pk, 2022. g. 24. jūn.
14:42) rakstīja:
